src/pages/PostDetails/PostDetails.jsx

```diff
@@ -1,26 +1,29 @@
 import { LikeOutlined } from '@ant-design/icons'
 import { IconText, Loading } from 'components'
-import { useEffect, useState } from 'react'
-import { useParams } from 'react-router-dom'
+import { useQuery } from 'react-query'
+import { useNavigate, useParams } from 'react-router-dom'
+import { postService } from 'services'
 
 const PostDetails = () => {
   const { postId } = useParams()
-  const [post, setPost] = useState()
-  const [loading, setLoading] = useState(true)
 
-  useEffect(() => {
-    fetch(`http://localhost:3000/posts/${postId}`)
-      .then((response) => response.json())
-      .then((data) => {
-        setPost(data)
-      })
-      .finally(() => setLoading(false))
-  }, [])
+  const {
+    isLoading,
+    data: post,
+    error,
+  } = useQuery(['post', postId], () => postService.fetchPost(postId))
+
+  const navigate = useNavigate()
+  const handleBackHome = () => navigate('/not-found')
+
+  if (error) {
+    handleBackHome()
+  }
 
   return (
     <>
-      {loading && <Loading />}
-      {!loading && (
+      {isLoading && <Loading />}
+      {!isLoading && (
         <>
           <h1>{post.title}</h1>
```

Comment:
Don't you need to use `?.` after `post` ? What with initial rendering? I'm not sure if object exists.